components/brave_shields/resources/panel/container.tsx

```diff
@@ -6,50 +6,91 @@ import * as React from 'react'
 
 import MainPanel from './components/main-panel'
 import TreeList from './components/tree-list'
-import { ViewType } from './state/component_types'
+import {
+  makeResourceInfoList,
+  ResourceInfo,
+  ResourceState,
+  ResourceType,
+  ViewType
+} from './state/component_types'
 import DataContext from './state/context'
 import styled from 'styled-components'
 import { getLocale } from '../../../common/locale'
+import { SiteBlockInfo } from './api/panel_browser_api'
 
 const Box = styled.div`
   position: relative;
 `
 
+function getResourceListForView (view: ViewType,
+                                 state: ResourceState,
+                                 siteBlockInfo: SiteBlockInfo): ResourceInfo[] {
+  if (view === ViewType.AdsList && state === ResourceState.Blocked) {
+    return makeResourceInfoList(siteBlockInfo.adsList, ResourceType.Ad, state)
+  }
+  if (view === ViewType.HttpsList && state === ResourceState.Blocked) {
+    return makeResourceInfoList(siteBlockInfo.httpRedirectsList,
+                                ResourceType.Http,
+                                state)
+  }
+  if (view === ViewType.ScriptsList && state === ResourceState.Blocked) {
+    return makeResourceInfoList(siteBlockInfo.blockedJsList,
+                                ResourceType.Http,
+                                state)
+  }
+  if (view === ViewType.ScriptsList && state === ResourceState.AllowedOnce) {
+    return makeResourceInfoList(siteBlockInfo.allowedJsList,
+                                ResourceType.Http,
+                                state)
+  }
+
+  return []
+}
+
 function Container () {
   const { siteBlockInfo, viewType } = React.useContext(DataContext)
-  const detailView = viewType !== ViewType.Main && siteBlockInfo
-
-  const renderDetailView = () => {
-    if (viewType === ViewType.AdsList && detailView) {
-      return (<TreeList
-        data={siteBlockInfo?.adsList}
-        totalBlockedCount={siteBlockInfo?.adsList.length}
-        blockedCountTitle={getLocale('braveShieldsTrackersAndAds')}
-      />)
-    }
+  const shouldShowDetailView = viewType !== ViewType.Main && siteBlockInfo
 
-    if (viewType === ViewType.HttpsList && detailView) {
-      return (<TreeList
-        data={siteBlockInfo?.httpRedirectsList}
-        totalBlockedCount={siteBlockInfo?.httpRedirectsList.length}
-        blockedCountTitle={getLocale('braveShieldsConnectionsUpgraded')}
-      />)
+  const blockedList = siteBlockInfo ?
+    getResourceListForView(viewType, ResourceState.Blocked, siteBlockInfo) : []
```

Comment:
It would be more manageable and efficient to apply transformations on siteBlockInfo at the source where the data is being obtained. The best solution would be to perform the transformation in the useSiteBlockInfoData hook. If possible, it would be ideal to handle the transformation on the C++ side.

Reply:
removed this code